addons/website_sale/views/templates.xml

```diff
@@ -2291,12 +2291,12 @@
             >
                 <!-- We don't allow public users to have multiple delivery addresses. -->
                 <t t-if="is_invoice">
-                    <t t-set="new_address_href" t-valuef="/shop/address?address_type=billing"/>
+                    <t t-set="new_address_href" t-valuef="/shop/address?address_type=billing&amp;use_delivery_as_billing={{use_delivery_as_billing}}&amp;partner_id={{order.partner_id.id}}"/>
```

Comment:
Providing the `use_delivery_as_shipping` seems right, but providing the partner id is wrong ...
Giving a partner to `/shop/address` means you're editing an existing address which isn't the case here, those href are providing ways to create a new address, not update the existing one set on the cart.

About the display of those fields only if the address is the main address of the customer (e.g. the main customer on the cart), as far as I know, most ecommerce localizations didn't even consider the possibility of having multiple billing addresses.  Nevertheless, it makes sense to hide those fields from additional addresses because new addresses are child addresses of the customer main address, and since those fields are `_commercial_fields` (see related logic on `res.partner` model), they are not meant to be set on child addresses (it somehow works, but would be reset if the field is modified on the main address).

The address validation (and client-side js logic) was adapted with #184612 so that the address were considered valid, but there might be some issues left in some existing localisations like the CO one.